clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -2555,6 +2561,32 @@ LogicalResult GetMemberOp::verify() {
   return mlir::success();
 }
 
+//===----------------------------------------------------------------------===//
+// GetRuntimeMemberOp Definitions
+//===----------------------------------------------------------------------===//
+
+LogicalResult GetRuntimeMemberOp::verify() {
+  auto recordTy = getAddr().getType().getPointee().dyn_cast<StructType>();
+  if (!recordTy) {
+    emitError() << "expected pointer to a record type";
```

Comment:
This should be enforced directly in tablegen by requiring the pointer to be a "pointer to a struct", see `ExceptionInfoPtr` for an example.

Reply:
Updated.